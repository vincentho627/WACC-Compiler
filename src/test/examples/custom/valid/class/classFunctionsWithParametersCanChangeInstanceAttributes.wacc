# Simple Class Function with Parameters

begin
  class Vehicle is
    private int name ;
    public int tyres 
    public int getName() is 
      return name
    end
    public int getNumber() is
      return tyres
    end
    public int setNumber(int offset) is
      tyres = tyres + offset ;
      return tyres
    end
  end
  Vehicle bus = new Vehicle(7, 6) ;
  Vehicle car = new Vehicle(6, 4) ;
  println bus.tyres ;
  println car.tyres ;
  int v = call bus.setNumber(5) ;
  println v ;
  println bus.tyres 
end








