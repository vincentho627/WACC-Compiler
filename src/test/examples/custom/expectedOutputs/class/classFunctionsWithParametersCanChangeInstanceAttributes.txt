6
4
11
11